### ISSUE TITLE
Implement editor undo stack serialization

### ISSUE THREAD

**Issue by joaomoreno** [Microsoft/vscode#7536]
Related to #7169, #7535

Similarly to the view state, we need a way to serialize, and later deserialize, a model's undo stack.

**Comment by alexandrudima** [comment]
Trying for July...

**Comment by rebornix** [comment]
@alexdima I think this is already done with your implementation of undo/redo persistence?

**Comment by joaomoreno** [comment]
He didn't mention which July. 😆

**Comment by alexdima** [comment]
No, the undo/redo is currently kept in memory, it is not saved to disk.

**Comment by martin-braun** [comment]
> Trying for July...

This didn't age so well, oh well ... 😅

**Comment by huachuman** [comment]
My undos don't persist across sessions... even though Restore undo stack is checked. Does this feature work for anyone?
